src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets

```diff
@@ -143,6 +156,61 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <Target Name="_ReplaceFingerprintedBlazorJsForPublish" DependsOnTargets="ProcessPublishFilesForWasm" Condition="'$(WasmBuildingForNestedPublish)' != 'true' and '$(BlazorFingerprintBlazorJs)' == 'true'">
+    <ItemGroup>
+      <_BlazorJSStaticWebAsset Include="@(StaticWebAsset)" Condition="'%(FileName)' == '%(_BlazorJSFile.FileName)'" />
+      <_BlazorJSPublishCandidate Include="%(_BlazorJSStaticWebAsset.RelativeDir)%(_BlazorJSStaticWebAsset.FileName).%(_BlazorJSStaticWebAsset.Fingerprint)%(_BlazorJSStaticWebAsset.Extension)" />
+      <_BlazorJSPublishCandidate>
+        <RelativePath>_framework/$([System.IO.Path]::GetFileNameWithoutExtension('%(Filename)'))%(Extension)</RelativePath>
+      </_BlazorJSPublishCandidate>
+    </ItemGroup>
+
+    <DefineStaticWebAssets
+      CandidateAssets="@(_BlazorJSPublishCandidate)"
+      FingerprintCandidates="true"
+      FingerprintPatterns="@(_BlazorJSFingerprintPattern)"
+      SourceId="$(PackageId)"
+      SourceType="Computed"
+      AssetKind="All"
+      AssetMergeSource="$(StaticWebAssetMergeTarget)"
+      AssetRole="Primary"
+      AssetTraitName="WasmResource"
+      AssetTraitValue="boot"
+      CopyToOutputDirectory="Never"
+      CopyToPublishDirectory="PreserveNewest"
+      ContentRoot="%(_BlazorJSStaticWebAsset.ContentRoot)"
+      BasePath="%(_BlazorJSStaticWebAsset.BasePath)"
+    >
+      <Output TaskParameter="Assets" ItemName="_BlazorJSPublishStaticWebAssets" />
+    </DefineStaticWebAssets>
+    <DefineStaticWebAssetEndpoints
+      CandidateAssets="@(_BlazorJSPublishStaticWebAssets)"
+      ExistingEndpoints="@(StaticWebAssetEndpoint)"
+      ContentTypeMappings="@(StaticWebAssetContentTypeMapping)"
+    >
+      <Output TaskParameter="Endpoints" ItemName="_BlazorJSPublishStaticWebAssetsEndpoint" />
+    </DefineStaticWebAssetEndpoints>
+    <PropertyGroup>
+      <_BlazorJSStaticWebAssetFullPath>@(_BlazorJSStaticWebAsset->'%(FullPath)')</_BlazorJSStaticWebAssetFullPath>
+    </PropertyGroup>
+    <ItemGroup>
+      <_BlazorJSStaticWebAsset Include="@(StaticWebAsset)" Condition="'%(AssetTraitName)' == 'Content-Encoding' and '%(RelatedAsset)' == '$(_BlazorJSStaticWebAssetFullPath)'" />
+    </ItemGroup>
+    <FilterStaticWebAssetEndpoints Condition="'@(_BlazorJSStaticWebAsset)' != ''"
+      Endpoints="@(StaticWebAssetEndpoint)"
+      Assets="@(_BlazorJSStaticWebAsset)"
+      Filters=""
+    >
+      <Output TaskParameter="FilteredEndpoints" ItemName="_BlazorJSEndpointsToRemove" />
+    </FilterStaticWebAssetEndpoints>
+    <ItemGroup>
+      <StaticWebAsset Remove="@(_BlazorJSStaticWebAsset)" />
+      <StaticWebAsset Include="@(_BlazorJSPublishStaticWebAssets)" />
+      <StaticWebAssetEndpoint Remove="@(_BlazorJSEndpointsToRemove)" />
+      <StaticWebAssetEndpoint Include="@(_BlazorJSPublishStaticWebAssetsEndpoint)" />
+    </ItemGroup>
+  </Target>
+
```

Comment:
This part is confusing me a bit, do we rewrite blazor.js in any way during this whole process? If not, I'm not sure why this is needed at all.

Reply:
It's the same issue we have with runtime assets. If we do hard fingerprint during build, the asset to promote for publish has identity without fingerprint (`bin/Debug/wwwroot/_framework/blazor.js`), but the file on disk has fingerprint (`bin/Debug/wwwroot/_framework/blazor.abcd.js`) and thus it complains that the file doesn't exist.

This is fixing by creating asset that has fingerprint in identity.

Maybe we can do soft fingerprint during build and hard only during publish. Any ideas?